File: php/lc00001-lc00100/lc00001/TwoSum.php
<?php

namespace leetcode\lc00001;

class TwoSum
{
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    public function twoSum($nums, $target)
    {
        $len = count($nums);
        $map = [];
        for ($i = 0; $i < $len; $i++) {
            $key = $target - $nums[$i];
            if (array_key_exists($key, $map)) {
                return [$map[$key], $i];
            }
            $map[$nums[$i]] = $i;
        }
        return [];
    }
}

File: golang/lc00001-lc00100/lc00001/two_sum.go
package lc00001

func twoSum(nums []int, target int) []int {
	for i, lenNums := 0, len(nums); i < lenNums; i++ {
		for j := i + 1; j < lenNums; j++ {
			if target == nums[i]+nums[j] {
				return []int{i, j}
			}
		}
	}

	return nil
}

// twoSumMap map
func twoSumMap(nums []int, target int) []int {
	var key int
	m := make(map[int]int, 0)
	for i, v := range nums {
		key = target - v
		if j, ok := m[key]; ok {
			return []int{j, i}
		}
		m[v] = i
	}

	return nil
}
